pkg/service/targetdeploymentsvc/create_test.go

```diff
@@ -0,0 +1,110 @@
+package targetdeploymentsvc
+
+import (
+	"context"
+	"testing"
+
+	"github.com/benbjohnson/clock"
+	"github.com/common-fate/common-fate/pkg/storage"
+	"github.com/common-fate/common-fate/pkg/targetgroup"
+	"github.com/common-fate/common-fate/pkg/types"
+	"github.com/common-fate/ddb"
+	"github.com/common-fate/ddb/ddbmock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/golang/mock/gomock"
+)
+
+func TestCreateTargetGroupDeployment(t *testing.T) {
+	type testcase struct {
+		name string
+		// database lookup return object (used to mock ErrTargetGroupDeploymentIdAlreadyExists)
+		mockGet *storage.GetTargetGroupDeployment
+		// database put object (used to mock ok response)
+		mockPut *targetgroup.Deployment
+		// input to CreateTargetGroupDeployment
+		give    types.CreateTargetGroupDeploymentRequest
+		wantErr error
+		want    *targetgroup.Deployment
+	}
+
+	testcases := []testcase{
+		{
+			name: "bad aws account number",
+			give: types.CreateTargetGroupDeploymentRequest{
+				AwsAccount: "123_bad_123",
+			},
+			wantErr: ErrInvalidAwsAccountNumber,
+		},
+		{
+			name: "existing deployment found",
+			mockGet: &storage.GetTargetGroupDeployment{
+				ID: "test1",
+				Result: targetgroup.Deployment{
+					ID: "test1",
+				},
+			},
+			give: types.CreateTargetGroupDeploymentRequest{
+				Id:         "test1",
+				AwsAccount: "123456789012",
+			},
+			wantErr: ErrTargetGroupDeploymentIdAlreadyExists,
+		},
+		{
+			name: "ok",
+			mockPut: &targetgroup.Deployment{
+				ID:         "test1",
+				AWSAccount: "123456789011",
+			},
+			give: types.CreateTargetGroupDeploymentRequest{
+				Id:         "test1",
+				AwsAccount: "123456789012",
+			},
+			want: &targetgroup.Deployment{
+				ID:         "test1",
+				AWSAccount: "123456789012",
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+
+			dbMock := ddbmock.New(t)
+
+			if tc.mockGet != nil {
+				// this is used to mock the db lookup for coverage of ErrTargetGroupDeploymentIdAlreadyExists
+				dbMock.MockQuery(tc.mockGet)
+			} else {
+				// this is used to mock s.DB.Put
+				dbMock.MockQueryWithErr(&storage.GetTargetGroupDeployment{}, ddb.ErrNoItems)
+			}
+			if tc.mockPut != nil {
+				ctx := context.Background()
+				dbMock.Put(ctx, tc.mockPut)
```

Comment:
This method returns an error so it needs to be checked 
if err != nil {
t.Fatal(err)
}